### PR TITLE
[Merged by Bors] - feat: Port CategoryTheory.Limits.Shapes.Multiequalizer

### PR DESCRIPTION
**Pull request by adamtopaz**
---
- [x] depends on: #2767
- [x] depends on: #2791 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2767
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by adamtopaz** [comment]
It looks like the `simpNF` linter still has a lot of complaints. Any help would be much appreciated!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2767~~
* ~~leanprover-community/mathlib4#2791~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean [hunk 1]
```diff
@@ -398,10 +398,7 @@ theorem hom_comp_ι (K₁ K₂ : Multifork I) (f : K₁ ⟶ K₂) (j : I.L) : f.
 #align category_theory.limits.multifork.hom_comp_ι
   CategoryTheory.Limits.Multifork.hom_comp_ι
 
-/-
-Porting note: Added simp lemma.
--/
-@[simp]
+-- @[simp] -- Porting note: simp can prove this
 lemma hom_w {K₁ K₂ : Multifork I} (f : K₁ ⟶ K₂) (b : I.L) : f.Hom ≫ K₂.ι b = K₁.ι b :=
```
[the remark this answers is not in the document]
Comment (adamtopaz) [reply Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean:402]: It seems that the dual lemma was not added in mathlib3 for some reason, whereas it's the lemma `hom_comp_\iota` above.
We should thus rename the dual lemma to be `\pi_comp_hom`.

[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean]: Done.

[the remark this answers is not in the document]
Comment (adamtopaz) [reply Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean]: I think you misunderstood what I meant. I was suggesting to delete the lemma in question here, but to rename the lemma called `hom_w` several lines below about morphisms of multicoforks.

[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean]: Done.

File: Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean [hunk 2]
```diff
@@ -59,37 +59,52 @@ variable {L R : Type w} {fst snd : R → L}
...
-  | _, _, _, hom.id X, f => f
-  | _, _, _, hom.fst b, hom.id X => Hom.fst b
-  | _, _, _, hom.snd b, hom.id X => Hom.snd b
-#align category_theory.limits.walking_multicospan.hom.comp CategoryTheory.Limits.WalkingMulticospan.Hom.comp
+/-- Composition of morphisms for `WalkingMulticospan`. -/
+def Hom.comp : ∀ {A B C : WalkingMulticospan fst snd} (_ : Hom A B) (_ : Hom B C), Hom A C
+  | _, _, _, Hom.id X, f => f
+  | _, _, _, Hom.fst b, Hom.id _ => Hom.fst b
+  | _, _, _, Hom.snd b, Hom.id _ => Hom.snd b
+#align category_theory.limits.walking_multicospan.hom.comp
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean:81]: Align statements should be on a single line. Also elsewhere.

File: Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean [hunk 3]
```diff
@@ -59,37 +59,52 @@ variable {L R : Type w} {fst snd : R → L}
...
+  id_comp := by
+    rintro (_ | _) (_ | _) (_ | _ | _) <;> rfl
+  comp_id := by
+    rintro (_ | _) (_ | _) (_ | _ | _) <;> rfl
+  assoc := by
+    rintro (_ | _) (_ | _) (_ | _) (_ | _) (_ | _ | _) (_ | _ | _) (_ | _ | _) <;> rfl
+
+/-
+Porting note: Add simp lemma.
+-/
```
[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean]: All done (along with inlining the aligns).

**Pull request closed without merging**